evap/evaluation/auth.py

```diff
@@ -1,83 +1,11 @@
-    """
-
-    field_name = "loginkey"
-
-    def __init__(self, get_response):
-        self.get_response = get_response
-
-    def __call__(self, request):
-        self.process_request(request)
-        response = self.get_response(request)
-        return response
-
-    def process_request(self, request):
-        # AuthenticationMiddleware is required so that request.user exists.
-        if not hasattr(request, 'user'):
-            raise ImproperlyConfigured(
-                "The Django remote user auth middleware requires the"
-                " authentication middleware to be installed.  Edit your"
-                " MIDDLEWARE_CLASSES setting to insert"
-                " 'django.contrib.auth.middleware.AuthenticationMiddleware'"
-                " before the RequestAuthMiddleware class.")
-
-        try:
-            key = int(request.GET[self.field_name])
-        except (KeyError, ValueError):
-            # If specified variable doesn't exist or does not convert to an int
-            # then return (leaving request.user set to AnonymousUser by the
-            # AuthenticationMiddleware).
-            return
-
-        # We are seeing this user for the first time in this session, attempt to authenticate the user.
-        user = auth.authenticate(request, key=key)
-
-        if user and not user.is_active:
-            messages.error(request, _("Inactive users are not allowed to login."))
-            return
-
-        # If we already have an authenticated user don't try to login a new user. Show an error message if another user
-        # tries to login with a URL in this situation.
-        if request.user.is_authenticated:
-            if user != request.user:
-                messages.error(request, _("Another user is currently logged in. Please logout first and then use the login URL again."))
-            return
-
-        if user and user.login_key_valid_until >= date.today():
-            # User is valid. Set request.user and persist user in the session by logging the user in.
-            request.user = user
-            auth.login(request, user)
-            messages.success(request, _("Logged in as %s.") % user.full_name)
-            # Invalidate the login key, but keep it stored so we can later identify the user that is trying to login and send a new link
-            user.login_key_valid_until = date.today() - timedelta(1)
-            user.save()
-        elif user:
-            # A user exists, but the login key is not valid anymore. Send the user a new one.
-            user.ensure_valid_login_key()
-            EmailTemplate.send_login_url_to_user(user)
-            messages.warning(request, _("The login URL is not valid anymore. We sent you a new one to your email address."))
-        else:
-            messages.warning(request, _("Invalid login URL. Please request a new one below."))
+from evap.evaluation.models import UserProfile
 
 
 class RequestAuthUserBackend(ModelBackend):
```

Comment:
the comment in the backend below has a reference to the middleware that you removed.